xwiki-platform-core/xwiki-platform-notifications/xwiki-platform-notifications-notifiers/xwiki-platform-notifications-notifiers-default/src/main/java/org/xwiki/notifications/notifiers/internal/UserEventDispatcher.java

```diff
@@ -116,165 +114,178 @@ public class UserEventDispatcher implements Runnable, Disposable, Initializable
     private EventStore events;
 
     @Inject
-    private Logger logger;
+    private RecordableEventDescriptorManager recordableEventDescriptorManager;
 
-    @Override
-    public void initialize() throws InitializationException
-    {
-        // Live events (event which just been produced) are put in the priority queue to be dealt with faster and we put
-        // events coming from some migration or restart in a larger queue but containing only the id to not flood the
-        // live events queue. If the priority queue is full (very very active wiki or some mistake in a script) events
-        // will fallback on the large queue. If both queue are full we just ignore the event and it will be picked up by
-        // the next missing pre filtered event run.
-        this.priorityQueue = new LinkedBlockingQueue<>(1000);
-        this.secondaryQueue = new LinkedBlockingQueue<>(100000);
-
-        // Start a background thread to filter and dispatch users events
-        Thread thread = new Thread(this);
-        thread.setName("User event dispatcher thread");
-        thread.setPriority(Thread.NORM_PRIORITY - 1);
-        thread.setDaemon(true);
-        thread.start();
-    }
+    @Inject
+    private RemoteObservationManagerConfiguration remoteObservation;
 
-    @Override
-    public void dispose() throws ComponentLifecycleException
-    {
-        // Mark the component as disposed
-        this.disposed = true;
+    @Inject
+    private ExecutionContextManager contextManager;
 
-        // Make sure to wake the thread up to stop it
-        wakeup();
-    }
+    @Inject
+    private Execution execution;
 
-    private void wakeup()
+    @Inject
+    private Logger logger;
+
+    private final ScheduledExecutorService scheduler = Executors.newSingleThreadScheduledExecutor();
+
+    private volatile boolean running;
+
+    /**
+     * Start the scheduler.
+     */
+    public void initialize()
     {
-        if (this.priorityQueue != null && this.priorityQueue.isEmpty()) {
-            // We don't need to wait since we just want to make sure that the queue is not empty
-            this.priorityQueue.offer(WAKEUP_EVENT);
-        }
+        // Schedule a job to regularly check if any event prefiltering was missed or a previous run failed
+        this.scheduler.scheduleWithFixedDelay(this::run, 0, 1, TimeUnit.HOURS);
```

Comment:
No reason to make that configurable?

Reply:
This is not new actually, what changed is just the way to do this scheduling.

Reply:
Right was defined in the listener previously. Ok so maybe a future improvment if needed then.

Reply:
Yes probably.